app/policies/project_policy.rb

```diff
@@ -0,0 +1,51 @@
+class ProjectPolicy
+  attr_reader :user, :project
+
+  def initialize(user, project)
+    @user = user
+    @record = project
+  end
+
+  def index?
+    true
+  end
+
+  def show?
+    user.team_leader? or @record.delivery_officer == @user
+  end
+
+  def create?
+    user.team_leader?
+  end
+
+  def new?
+    create?
+  end
+
+  def edit?
+    user.team_leader?
+  end
+
+  def update?
+    edit?
+  end
+
+  class Scope
+    def initialize(user, scope)
+      @user = user
+      @scope = scope
+    end
+
+    def resolve
+      if user.team_leader?
+        scope.all
+      else
+        scope.where(delivery_officer: user)
+      end
+    end
+
+    private
+
+    attr_reader :user, :scope
```

Comment:
I thought this class with the nesting and private accessors was a bit odd, but I can see Pundit expects it like this. Interesting!